src/main/java/hudson/security/LDAPSecurityRealm.java

```diff
@@ -830,50 +839,98 @@ public LdapUserDetails updateUserDetails(LdapUserDetails d) {
 
     @Override
     public GroupDetails loadGroupByGroupname(String groupname) throws UsernameNotFoundException, DataAccessException {
+        return loadGroupByGroupname(groupname, false);
+    }
+
+    @Override
+    public GroupDetails loadGroupByGroupname(String groupname, boolean fetchMembers) throws UsernameNotFoundException, DataAccessException {
         groupname = fixGroupname(groupname);
-        Set<String> cachedGroups;
+        GroupDetails cachedGroup;
         if (cache != null) {
-            final CacheEntry<Set<String>> cached;
+            final CacheEntry<GroupDetails> cached;
             synchronized (this) {
                 cached = groupDetailsCache != null ? groupDetailsCache.get(groupname) : null;
             }
             if (cached != null && cached.isValid()) {
```

Comment:
Should the value in the cache be invalid if it was created with `fetchMembers == false` and the current call to `loadGroupByGroupname` has `fetchMembers == true`?

Reply:
yes the cache should be invalid in that case